File: frontend/lib/models/equipment.dart
import 'project.dart';

class Equipment {
  final int? equipmentId;
  final String equipmentNumber;
  final int projectId;
  final Project? project;
  final bool isActive;

  Equipment({
    this.equipmentId,
    required this.equipmentNumber,
    required this.projectId,
    this.project,
    this.isActive = true,
  });

  factory Equipment.fromJson(Map<String, dynamic> json) {
    return Equipment(
      equipmentId: json['equipmentId'] as int?,
      equipmentNumber: json['equipmentNumber'] as String,
      projectId: json['projectId'] as int,
      project: json['project'] != null ? Project.fromJson(json['project']) : null,
      isActive: json['isActive'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'equipmentId': equipmentId,
      'equipmentNumber': equipmentNumber,
      'projectId': projectId,
      'isActive': isActive,
    };
  }
}

File: backend/EquipmentLogApi/Domain/Entities/Equipment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EquipmentLogApi.Domain.Entities;

public class Equipment
{
    [Key]
    public int EquipmentId { get; set; }

    [Required]
    [MaxLength(100)]
    public string EquipmentNumber { get; set; } = string.Empty;

    [Required]
    public int ProjectId { get; set; }

    [ForeignKey("ProjectId")]
    public Project? Project { get; set; }

    public bool IsActive { get; set; } = true;
}
